src/preprocessor.ts

```diff
@@ -1,30 +1,40 @@
-import { parse as parser, ParserOptions } from '@babel/parser';
-import { merge } from 'lodash';
-import { loadPartialConfig } from '@babel/core';
+import merge from 'deepmerge';
+import { parse as babelParser, ParserOptions } from '@babel/parser';
 import traverse, { NodePath } from '@babel/traverse';
-import {
-    ImportDeclaration,
-    isTSModuleDeclaration,
-} from '@babel/types';
-import { PrettierParserOptions, getCodeFromAst, getSortedNodes } from './utils';
+import { ImportDeclaration, isTSModuleDeclaration } from '@babel/types';
 
-export function preprocessor(code: string, options: PrettierParserOptions) {
-    const { importOrder, importOrderSeparation } = options;
+import { getCodeFromAst } from './utils/get-code-from-ast';
+import { getSortedNodes } from './utils/get-sorted-nodes';
+import { getBabelConfing } from './utils/get-babel-config';
+import { getParserPlugins } from './utils/get-parser-plugins';
+import { PrettierOptions } from './types';
+
+export function preprocessor(code: string, options: PrettierOptions) {
+    const {
+        importOrder,
+        importOrderSeparation,
+        parser: prettierParser,
+        experimentalBabelParserPluginsList = [],
+    } = options;
+
+    const plugins = getParserPlugins(prettierParser);
 
     const importNodes: ImportDeclaration[] = [];
 
-    const defaultConfig = {
+    const defaultConfig: ParserOptions = {
         sourceType: 'module',
-        plugins: ['typescript', 'jsx'],
-    } as ParserOptions;
-    const babelConfig = loadPartialConfig() as ParserOptions;
+        plugins: [...plugins, ...experimentalBabelParserPluginsList],
+    };
+    const babelConfig = getBabelConfing();
     const mergedOptions = merge(defaultConfig, babelConfig);
 
-    const ast = parser(code, mergedOptions);
+    const ast = babelParser(code, mergedOptions);
```

Comment:
I'm not 100% sure that we actually need `babelConfig` for the babel parser. Babel parser only needs options defined in `ParserOptions` type.

```typescript
export interface ParserOptions {
  allowImportExportEverywhere?: boolean;
  allowAwaitOutsideFunction?: boolean;
  allowReturnOutsideFunction?: boolean;
  allowSuperOutsideMethod?: boolean;
  allowUndeclaredExports?: boolean;
  errorRecovery?: boolean;
  sourceType?: "script" | "module" | "unambiguous";
  sourceFilename?: string;
  startLine?: number;
  plugins?: ParserPlugin[];
  strictMode?: boolean;
  ranges?: boolean;
  tokens?: boolean;
  createParenthesizedExpressions?: boolean;
}
```